examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,885 @@
+      "    for (int x = x_m; x <= x_M; x += 1)\n",
+      "    {\n",
+      "      #pragma omp simd aligned(f,g:32)\n",
+      "      for (int y = y_m; y <= y_M; y += 1)\n",
+      "      {\n",
+      "        if (y < 5 && g[x + 1][y + 1] != 0)\n",
+      "        {\n",
+      "          f[x + 1][y + 1] = f[x + 1][y + 1] + g[x + 1][y + 1];\n",
+      "        }\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  /* End section0 */\n",
+      "  gettimeofday(&end_section0, NULL);\n",
+      "  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 0., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 2., 0., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 2., 0., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 2., 0., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 2., 0., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 1., 0., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 1., 0., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],\n",
+       "      [0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "from sympy import And\n",
+    "from devito import Ne, Lt\n",
+    "\n",
+    "f.data[:,:] = np.eye(10)\n",
+    "\n",
+    "\n",
+    "g = Function(name='g', grid=grid)\n",
+    "g.data[:,:] = np.eye(10)\n",
+    "\n",
+    "ci = ConditionalDimension(name='ci', parent=y, condition=And(Ne(g, 0), Lt(y, 5)))\n",
+    "\n",
+    "op = Operator(Eq(f, f + g, implicit_dims=ci))\n",
+    "op.apply()\n",
+    "\n",
+    "print(op.ccode)\n",
+    "\n",
+    "assert (np.count_nonzero(f.data - np.diag(np.diagonal(f.data)))==0)\n",
+    "assert (np.count_nonzero(f.data) == 10)\n",
+    "assert np.all(f.data[np.nonzero(f.data[:5,:5])] == 2)\n",
+    "assert np.all(f.data[5:,5:] == np.eye(5))\n",
+    "\n",
+    "f.data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can see that `f` has been updated only for the first five columns with the `f+g` expression."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A `ConditionalDimension` can be also used at the construction time of a `Function`. Let's use `ci` from the previous cell to explicitly construct a function `h`. You will notice that in this case there is no need to pass argument `implicit_dims` to the `Equation` as the `ConditionalDimension` `ci` is now a dimension in `h`. `h` still has the size of the full domain, not the size of the domain that satisfies the condition."
```

Comment:
Hang on. Doesn't this belong to use case B ?

Reply:
Why? I consider use case B as the case where `factor` is used instead of `condition`. I do not understand why you say this.